File: backends/Contracts/Common.Infra/Cache/Redis/Handlers/Exists.cs
using D2.Contracts.Common;
using D2.Contracts.Common.App;
using H = D2.Contracts.Common.App.ICacheService.IExistsHandler;
using I = D2.Contracts.Common.App.ICacheService.ExistsInput;
using O = D2.Contracts.Common.App.ICacheService.ExistsOutput;

namespace Common.Infra.Cache.Redis.Handlers;

/// <inheritdoc cref="H"/>
public class Exists : BaseHandler<H, I, O>, H
{
    /// <summary>
    /// Initializes a new instance of the <see cref="Exists"/> class.
    /// </summary>
    ///
    /// <inheritdoc/>
    public Exists(IHandlerContext context) : base(context) { }

    /// <inheritdoc/>
    protected override ValueTask<D2Result<O?>> ExecuteAsync(
        I input,
        CancellationToken ct = default)
    {
        // TODO: Implement.
        return default;
    }
}

File: backends/node/shared/interfaces/src/caching/distributed/handlers/r/exists.ts
import type { IHandler } from "@d2/handler";

/** Input for checking if a key exists in the distributed cache. */
export interface ExistsInput {
  key: string;
}

/** Output for checking if a key exists in the distributed cache. */
export interface ExistsOutput {
  exists: boolean;
}

/** Handler for checking if a key exists in the distributed cache. */
export type IExistsHandler = IHandler<ExistsInput, ExistsOutput>;
